packages/patterns/src/partition.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Partition unordered data into buckets of similar total weight
+ *
+ * @example
+ * ```typescript
+ *  partition([1, 2, 3], 2, (i) => i)) // [[3], [2, 1]];
```

Comment:
Could ```(i) => i``` be the default?

Reply:
I think that's not interesting for numbers (other than examples and tests) usually you'd want to use the weight (as in test run time) to partition something else (test file name)